olx_parser.py

```diff
@@ -1,124 +1,181 @@
-                date_posted = format_date(soup_product_date_posted)
-                
-                soup_product_jpg = soup_product.find('img').get('src')                  
-                soup_product_title = soup_product.find('h4', class_='css-1juynto').text
-                soup_product_text = soup_product.find('div', class_='css-1t507yq er34gjf0').text
-        
-        except(requests.RequestException, ValueError, AttributeError):
-            print('error')
+            card_url = div.find('a').get('href')
+            if card_url:
+                if 'https://' in card_url:
+                    adt_urls.append(card_url)
+                else:
+                    adt_urls.append(f'https://www.olx.kz{card_url}')
+        except Exception as err:
+            logger.error('error extract_adt_urls', exc_info=err)
             continue
+    print(adt_urls)
+    return (adt_urls)
+
+
+def parser_adt(adv_html: str) -> Adv | None:
+    adv_html = BeautifulSoup(adv_html, 'lxml')
+    try:
+        soup_product_number_id = adv_html.find('span', class_='css-12hdxwj er34gjf0').text # ID объявления
+        soup_product_name_salesman = adv_html.find('h4', class_='css-1lcz6o7 er34gjf0').text # Ник продавца
+        soup_product_date_registered = adv_html.find('div', class_='css-16h6te1 er34gjf0').text 
+        date_registered = parse_register_date(soup_product_date_registered) # Дата регистрации аккаунта
+        soup_product_date_of_last_visit = adv_html.find('span', class_='css-1t0qnkx').text
+        date_of_last_visit = parse_last_online_date(soup_product_date_of_last_visit) # Время последнего посещения сайта
+        soup_product_date_posted = adv_html.find('span', class_='css-19yf5ek').text
+        date_posted = parse_published_date(soup_product_date_posted) # Дата размещения объявления
+        soup_product_jpg = adv_html.find('img').get('src') # Ссылка на фотографию объявления
+        soup_product_title = adv_html.find('h4', class_='css-1juynto').text # Название объявления
+        soup_product_text = adv_html.find('div', class_='css-1t507yq er34gjf0').text # Описание объявления   
+    except(requests.RequestException, ValueError, AttributeError) as err:
+        logger.error(f'{"_"*51}error adt{"_"*51}',  exc_info=err)
+        return None
+    adv = Adv(
+            id=soup_product_number_id,
+            ads_name=soup_product_title,
+            ads_content=soup_product_text,
+            seller_name=soup_product_name_salesman,
+            date_registered=date_registered,
+            date_of_last_visit=date_of_last_visit,
+            date_posted=date_posted,
+            picture=soup_product_jpg
+        )
+    return adv
+
+def parse_register_date(date_time: str) -> datetime:
+    #  Дата регистрации пользователя
+    date_time = date_time.split(' ')
+    result = f'{translate_month_to_en(date_time[2])}, {date_time[3]}'
+    return datetime.strptime(result, '%B, %Y')
```

Comment:
вот для этого нужно сразу интовый месяц получать из translate_month_to_en
```suggestion
    dt_elements = date_time.split(' ')
    month = translate_month_to_en(dt_elements[2])
    year = int(dt_elements[3])
    return datetime(year, month, day=1)
```